pennylane/ftqc/lattice.py

```diff
@@ -0,0 +1,189 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# pylint: disable=protected-access, inconsistent-return-statements
+"""
+This file defines classes and functions for creating lattice objects that store topological
+connectivity information for measurement-based quantum computing (MBQC) and fault-tolerant
+quantum computing (FTQC).
```

Comment:
```suggestion
This file defines classes and functions for creating lattice objects that store qubit topological
connectivity information.
```

We don't need the class and functions to be FTQC aware --- we should think of it as a building block only; something that produces a lattice with the correct structure. How we use that structure isn't something we currently need here.